### PR TITLE
Add POST functionality to add comments

### PR DESCRIPTION
**Pull request by arjavibahety**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -19,14 +19,38 @@
...
+    response.getWriter().println(comments);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = getParameter(request, "comment-input", "");
+    comments.add(comment);    
+    response.sendRedirect("/index.html");
+
+    // response.setContentType("application/json;");
```
Comment (kevinluvian) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:41]: remove comments

File: portfolio/src/main/webapp/js/scripts.js [hunk 2]
```diff
@@ -47,5 +47,7 @@
 async function getData() {
     const response = await fetch('/data');
     const responseText = await response.text();
-    document.getElementById('greeting-container').innerHTML = responseText;
+    console.log(response);
```
Comment (kevinluvian) [inline portfolio/src/main/webapp/js/scripts.js:50]: remove console log

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -71,6 +71,11 @@
               >Interests</a
             >
           </li>
+          <li class="nav-item">
+            <a class="nav-link js-scroll-trigger" href="#comments"
+              >Comments</a
```
Comment (kevinluvian) [inline portfolio/src/main/webapp/index.html:76]: I just noticed this.
I think it is better to have the ">" on the same line as the "<" to make it easier to read. e.g:
```
<a class="nav-link js-scroll-trigger" href="#comments">
    Comments
</a>
```
or if you cant, we can put the attributes into multiple lines, like so:
```
<a class="nav-link js-scroll-trigger" 
     href="#comments">
    Comments
</a>
```
I feel that this is cleaner, what do you think?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -19,14 +19,38 @@
...
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Arjavi!</h1>");
+    response.getWriter().println(comments);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = getParameter(request, "comment-input", "");
```
Comment (kevinluvian) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:37]: it is better to make "comment-input" a private static final constant variable

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -19,14 +19,38 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    ArrayList<String> comments = new ArrayList<>();
```
Comment (kevinluvian) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:27]: the indentation is not aligned here

File: portfolio/src/main/webapp/index.html (hunk 3 above)
Comment (arjavibahety) [reply portfolio/src/main/webapp/index.html]: Yes, definitely!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -19,14 +19,36 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  ArrayList<String> comments = new ArrayList<>();
```
Comment (kevinluvian) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:27]: its a good practice to make variable private by default

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -19,14 +19,36 @@
...
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Arjavi!</h1>");
+    response.getWriter().println(comments);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = getParameter(request, COMMENT_INPUT, "");
```
Comment (kevinluvian) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:38]: just wondering, but are you going to allow empty string as comment?
how about spaces " "?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -24,26 +24,29 @@
...
   }
-
+    
+  /**
+   * @return the json conversion of comments list
+   */
   private String convertToJson(ArrayList<String> arr) {
     String json = "{";
-    
+
```
Comment (kevinluvian) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:49]: there is extra whitespace here

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 9]
```diff
@@ -54,4 +57,17 @@ private String convertToJson(ArrayList<String> arr) {
...
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+
```
Comment (kevinluvian) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:71]: extra line here is not necessary

File: .gitignore [hunk 10]
```diff
@@ -1,2 +1,4 @@
 target/
 .idea
+config.js
```
Comment (kevinluvian) [inline .gitignore:3]: this shouldn't be in this PR.
if its hard to remove it, we can keep it

**Pull request merged**